File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/CPP/BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if( prices.size() <2)
            return 0;
            
        int result=0;
        int min=prices[0];
        for( int i=1; i<prices.size(); i++){
            if( prices[i]<min){
                min=prices[i];
                continue;
            }
            else if( prices[i]-min > result)
                result=prices[i]-min;
            else
                ;
        }
        
        return result;
    }
};

File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Python/BestTimetoBuyandSellStock.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        if len(prices) < 2: return 0
        lowest = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] < lowest:
                lowest = prices[i]
            if prices[i] - lowest > profit:
                profit = prices[i] - lowest
                
        return profit 
